crypto/dilithium/p_pqdsa_test.cc

```diff
@@ -403,8 +1023,14 @@ TEST_P(PQDSAParameterTest, KAT) {
 
     // Generate key pair from seed xi and assert that public and private keys
     // are equal to expected values from KAT
-    if (name == "MLDSA65") {
-      ASSERT_TRUE(ml_dsa_65_keypair_internal(pub.data(), priv.data(), xi.data()));
+    if (name == "MLDSA44") {
```

Comment:
Can the KAT Test use the EVP API like the following tests, or it's intended to use the internal APIs?

Reply:
If we would like the KATs to use the EVP APIs, we would need to add new EVP methods for `sign_internal` and `verify_internal`. Those new methods would allow the randomness to be supplied as an additional argument, and call the internal functions rather than the external. Given that these internal methods are for testing/validation purposes only, there is no use for them within the EVP APIs. 

These internal functions were not designed for any other use-cases than testing, see FIPS 204 page 22 section 6.

```
Other than for testing purposes, the interfaces for key generation and signature generation specified
in this section should not be made available to applications, as any random values required for key
generation and signature generation shall be generated by the cryptographic module. 
```